src/ast.rs

```diff
@@ -5843,7 +5843,7 @@ where
                 Ok(())
             }
             Expression::PackExpansion(ref pack) => {
-                pack.demangle(ctx, scope)?;
+                pack.demangle_as_subexpr(ctx, scope)?;
```

Comment:
We should add the test case from this second commit's message to the actual test suite so that we know if we regress this behavor.

Reply:
Okay, I added it to the test suite!